components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/util/DCRConfigUtils.java

```diff
@@ -0,0 +1,267 @@
+
+        return DCRDataHolder.getInstance().getConfigurationManager();
+    }
+
+    private static void validateMandateSSA (DCRConfiguration dcrConfiguration) throws DCRMClientException {
+
+        if (dcrConfiguration.getMandateSSA() != null && dcrConfiguration.getMandateSSA() &&
+                StringUtils.isBlank(dcrConfiguration.getSsaJwks())) {
+            throw handleClientException(DCRMConstants.DCRConfigErrorMessage.ERROR_CODE_SSA_JWKS_REQUIRED);
+        }
+    }
+
+    /**
+     * Get DCR configuration by tenant domain.
+     * If there is a resource available for the tenant with the given resource type and resource name,
+     * it will override the server configuration.
+     * @return DCRConfiguration.
+     * @throws DCRMServerException DCRMServerException.
+     */
+    public static DCRConfiguration getDCRConfiguration() throws DCRMServerException {
+
+        try {
+//            tenantDomain is resolved inside getResource() method.
+            Resource resource = getResource(DCR_CONFIG_RESOURCE_TYPE_NAME, DCR_CONFIG_RESOURCE_NAME);
+            DCRConfiguration dcrConfiguration = getDCRServerConfiguration();
+            if (resource != null) {
+                overrideDCRServerConfigsWithDCRResourceConfig(resource, dcrConfiguration);
+            }
+
+            return dcrConfiguration;
+        } catch (ConfigurationManagementException e) {
+            throw handleServerException(ERROR_CODE_DCR_CONFIGURATION_RETRIEVE, e);
+        }
+    }
+
+    /**
+     * Configuration Management API returns a ConfigurationManagementException with the error code CONFIGM_00017 when
+     * resource is not found. This method wraps the original method and returns null if the resource is not found.
+     *
+     * @param resourceTypeName Resource type name.
+     * @param resourceName     Resource name.
+     * @return Retrieved resource from the configuration store. Returns {@code null} if the resource is not found.
+     * @throws ConfigurationManagementException exception
+     */
+    private static Resource getResource(String resourceTypeName, String resourceName)
+            throws ConfigurationManagementException {
+
+        try {
+
+            return getConfigurationManager().getResource(resourceTypeName, resourceName);
+        } catch (ConfigurationManagementException e) {
+            if (ERROR_CODE_RESOURCE_DOES_NOT_EXISTS.getCode().equals(e.getErrorCode()) ||
+                    ERROR_CODE_RESOURCE_TYPE_DOES_NOT_EXISTS.getCode().equals(e.getErrorCode())) {
+
+                return null;
+            } else {
+                throw e;
+            }
+        }
+    }
+
+    /**
+     * Get the DCR configuration from the server configuration.
+     *
+     * @return DCRConfiguration The DCR configuration.
+     */
+    private static DCRConfiguration getDCRServerConfiguration() throws DCRMServerException {
+
+        DCRConfiguration dcrConfiguration = new DCRConfiguration();
```

Comment:
i feel line 148-168 can be make as a seperate util method as same logic uses in 'overrideDCRServerConfigsWithDCRResourceConfig' also. U cn pass values from properties/resources as input params for util method.